src/Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -1540,17 +1540,47 @@ public VoidObject VisitWhileStmt(Stmt.While stmt)
                     }
                     else if (left is int && right is int)
                     {
-                        int leftNumber = leftConvertible!.ToInt32(CultureInfo.InvariantCulture);
-                        int rightNumber = rightConvertible!.ToInt32(CultureInfo.InvariantCulture);
+                        int leftInt = (int)left;
+                        int rightInt = (int)right;
 
-                        return leftNumber + rightNumber;
+                        return leftInt + rightInt;
                     }
                     else if (left is int or long && right is int or long)
                     {
-                        long leftNumber = leftConvertible!.ToInt64(CultureInfo.InvariantCulture);
-                        long rightNumber = rightConvertible!.ToInt64(CultureInfo.InvariantCulture);
+                        long leftLong = leftConvertible!.ToInt64(CultureInfo.InvariantCulture);
+                        long rightLong = rightConvertible!.ToInt64(CultureInfo.InvariantCulture);
 
-                        return leftNumber + rightNumber;
+                        return leftLong + rightLong;
+                    }
+                    else if (left is uint && right is uint)
+                    {
+                        uint leftUint = (uint)left;
+                        uint rightUint = (uint)right;
+
+                        return leftUint + rightUint;
+                    }
+                    else if (left is int && right is uint)
+                    {
+                        int leftInt = (int)left;
+                        uint rightUint = (uint)right;
+
+                        return leftInt + rightUint;
+                    }
+                    else if (left is uint && right is int)
+                    {
+                        uint leftUint = (uint)left;
+                        int rightInt = (int)right;
+
+                        if (expr.TypeReference.ClrType == typeof(uint))
+                        {
+                            // TODO: Is this safe? Do we perform enough typechecks elsewhere to feel confident enough about it?
+                            // TODO: If it's good, we should do something similar on other arm(s) as well.
+                            return (uint)(leftUint + rightInt);
+                        }
+                        else
+                        {
+                            return leftUint + rightInt;
+                        }
                     }
```

Comment:
This arm (`uint + int`) might not be relevant right now, at least not for variables:

```
> var i: uint = 1;
> var j: int = -1;
> var k = i + j;
[line 1] Error at '+': Unsupported + operand types: 'uint' and 'int'
```

Will check with integer literals next.